src/shared/utils.js

```diff
@@ -23,6 +23,23 @@ export const Utils = {
     return ["win"].includes(os);
   },
 
+  connectingDelay() {
+    return 2500;
+  },
+
+  // Hack to mitigate FXVPN-217 and FXVPN-222.
+  // When transitioning from StateFirefoxVPNConnecting to StateFirefoxVPNEnabled,
+  // the client enters StateOnPartial wherein routing table updates can cause
+  // long page reloads or even timeouts.
+  // To offset this, we introduce a delay in the UI and before reloading tabs,
+  // allowing the OS a buffer in which to update routing tables.
+  delayToStateEnabledNeeded(currentState, newState) {
+    if (!currentState) {
+      return false;
+    }
+    return currentState === "Connecting" && newState === "Enabled";
+  },
```

Comment:
I don't think utils the the place for this to live. This is very much only relevant to the webExt Controller. 

IMHO: WebExt controller should be responsible for delaying the state. 
You could make the argument that webExtController should only set "Enabled" once it is ready. 

If we have each consumer have to implement this, one might not do that, already be "Enabled" while the other one is "Connecting". 


So consider either:
- Changing handleClientStateChanges to wait.
- Create a new propertytype "delayed" that in the pipe transparently applys` data --> |delay| --> consumer `

```js 
// Not tested just wrote here in gh.

export class ExtensionController extends Component {
  get state() {
    return delayedProperty(this.#mState, (currentState,newState)=>{
      if (!currentState) {
        return false;
      }
      return currentState === "Connecting" && newState === "Enabled";
    }, 1000);
  }
}

// property.js
export const delayedProperty = (bindable, delayIf = ()=>true, delayMS=100)=>{
  const inner = property(bindable.value);
  bindable.subscribe((newValue) => {
    if(!delayIf(inner.value,newValue)){
      inner.value=newValue;
      return;
    }
    setTimeout(()=>{
      inner.value=newValue;
    },delayMS);
  })
}

```

Reply:
> WebExt controller should be responsible for delaying the state.

Agree this would be better. I considered this too but ultimately decided to implement it on an as-needed basis in order to _not_ delay the change for the RequestHandler. Totally agree this could get very messy and hard to keep track of, especially if other UI elements need to reflect the change from connecting->enabled in the future. Going to mull this over a bit!